### PR TITLE
FYLE-2gg3hqk: Removed inconsistency in mobile app while unlinking card from an expense

### PR DESCRIPTION
**Pull request by Dimple16**
### What is the change in behavior?
- Previously, on clicking the `:` button, we showed the mandatory fields to be entered
- In the case of the Web app, we don't require to fill mandatory fields for Unlinking card details or dismiss/marking them as personal
- To fix this inconsistency in the mobile app, after this change:
    - On clicking the dot menu, we show the options available to the user
    - If the user wants to split an expense, they will have to fill the mandatory fields
    - Users will no longer have to fill mandatory fields to unlink card details or dismiss/mark as personal

https://user-images.githubusercontent.com/31147415/179176001-3d1941b3-5030-41f1-abc3-3ecf78e9dad5.mov

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -838,93 +838,114 @@ export class AddEditExpensePage implements OnInit {
...
+      if (invalidElement) {
+        invalidElement.scrollIntoView({
+          behavior: 'smooth',
+        });
+      }
+    }
+  }
+
   async moreActions() {
-    if (this.fg.valid) {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:842]: Change - Removed this `if condition`

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -838,93 +838,114 @@ export class AddEditExpensePage implements OnInit {
...
+        this.actionSheetButtons.push({
+          text: 'Unlink Card Details',
+          handler: () => {
+            this.unlinkCorporateCardExpense();
+          },
         });
-        await actionSheet.present();
-      });
-    } else {
-      this.fg.markAllAsTouched();
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:917]: Change - Moved this to a separate method - `showFormValidationErrors()`

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -838,93 +838,114 @@ export class AddEditExpensePage implements OnInit {
...
+      const orgSettings = res.orgSettings$;
+      const areCostCentersAvailable = res.costCenters.length > 0;
+      const areProjectsAvailable = orgSettings.projects.enabled && res.projects.length > 0;
+      const projectField = res.txnFields.project_id;
+
+      this.actionSheetButtons = [
+        {
+          text: 'Split Expense By Category',
+          handler: () => {
+            if (this.fg.valid) {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:879]: Change - Show validation errors - Only while splitting expenses

**Comment by Dimple16** [comment]
**Note: I have removed the toast message shown in the video above cc @Julias0**

**Review by Julias0 — APPROVED** [review]

**Pull request merged**